### ISSUE TITLE
Log in with different cases for IDIR creates multiple user records

### ISSUE THREAD

**Issue by basilv** [bcgov/nr-forest-client-api#77]
**Describe the Bug**
If I log in twice in IDIR, once using BVANDEGR and once using lower-case bvandegr, then two user records get created by lambda_function.py (because the check for existing user is based on case-sensitive username).

Access is only granted to BVANDEGR so this is bad. 

**Expected Behaviour**

**Actual Behaviour**

** Steps To Reproduce**
Steps to reproduce the behaviour:

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by basilv** [comment]
This might not actually be a defect, I got confused testing locally. It is possible (likely) that IDIR is always returning a consistent case (upper?)

**Comment by basilv** [comment]
Hmm, I might be wrong about the bug, I keep forgetting that the auth behavior happens in dev and not locally. So the defect I created might not be an issue. What happened locally is that I granted access to bvandegr using lower-case, which auto-created a corresponding user different from the existing BVANDEGR. So this suggests that not only is Nick's fix important, we need to look at the logic to find an existing user in the db to ensure it is a case insensitive search (or always use upper case).

So there is very likely something to fix. I think to be safe, lambda_function.py should force user name to upper case for extra clarity.
